### PR TITLE
Add hidden command to destroy runners

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
...without filesystem side effects 🤧 

Closes https://github.com/iterative/cml/issues/1138 along with https://github.com/iterative/cml/pull/1225.

### CODE REVIEW

File: cmd/leo/destroyrunner/destroyrunner.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	cmd.Flags().StringVar(&o.Name, "name", "", "needed for Google Cloud runners") // FIXME: it shouldn't
+
+	return cmd
+}
+
+func (o *Options) Run(cmd *cobra.Command, args []string, cloud *common.Cloud) error {
+	d := &schema.ResourceData{}
+
+	d.Set("region", cloud.Region)
+	d.Set("name", o.Name)
```
Comment (0x2b3bfa0) [inline cmd/leo/destroyrunner/destroyrunner.go:45]: Thanks, linter! Let's check for errors that will never happen:

```suggestion
	for key, value := map[string]string{
		"region": cloud.Region,
		"name": o.Name
	} {
		if err := d.Set(key, value); err != nil {
			return err
		}
	}
```

**Review by omesser — APPROVED** [review]
⚡

**Comment by dacbd** [comment]
@0x2b3bfa0 complete enough for some testing?

**Comment by 0x2b3bfa0** [comment]
@dacbd, tested and ready for peer review 🎉

**Review by tasdomas — APPROVED** [review]

File: cmd/leo/destroyrunner/destroyrunner.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	}
+
+	d := schema.TestResourceDataRaw(&testing.RuntimeT{}, s, r)
+	d.SetId(args[0])
+
+	switch cloud.Provider {
+	case common.ProviderAWS:
+		return aws.ResourceMachineDelete(context.TODO(), d, nil)
+	case common.ProviderGCP:
+		return gcp.ResourceMachineDelete(context.TODO(), d, nil)
```
[the remark this answers is not in the document]
Comment (0x2b3bfa0) [reply cmd/leo/destroyrunner/destroyrunner.go]: Yes, but they are all handled by the “machine” resource, and some of them left in the cloud to be reused. 🙊

**Review by tasdomas — APPROVED** [review]

**Pull request merged**